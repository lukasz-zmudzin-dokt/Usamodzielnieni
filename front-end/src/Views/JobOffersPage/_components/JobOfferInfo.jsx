import React from 'react';
import { Row, Col, Button } from 'react-bootstrap';
import { IndexLinkContainer } from 'react-router-bootstrap';

const JobOfferInfo = ({ offer, ...rest }) => {
    const getAddressString = (offer) => "TODO";
  
    return (
      <Row {...rest}>
        <Col>{offer.title}</Col>
        <Col>{offer.companyName}</Col>
        <Col>{getAddressString(offer)}</Col>
        <Col>
          <IndexLinkContainer to={`/jobOffers/${offer.id}`}>
            <Button>Pokaż szczegóły</Button>
          </IndexLinkContainer>
        </Col>
      </Row>
    )
}

export default JobOfferInfo;