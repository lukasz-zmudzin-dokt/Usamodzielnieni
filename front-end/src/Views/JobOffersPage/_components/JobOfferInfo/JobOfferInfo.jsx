import React from "react";
import { Row, Col, Button } from "react-bootstrap";
import { IndexLinkContainer } from "react-router-bootstrap";
import { DetailsItem } from "components";
import proxy from "config/api";

const JobOfferInfo = ({ context, offer, ...rest }) => {
  console.log(offer);
  return (
    <Row {...rest}>
      <Col>
        <h5>{offer.title}</h5>
        <Row>
          {offer.companyLogo ? (
            <DetailsItem md="3" xl="3" label="Logo firmy">
              <img
                className="JobOffer__logo"
                src={`${proxy.plain}${offer.companyLogo}`}
              />
            </DetailsItem>
          ) : null}
          <DetailsItem md="3" xl="3" label="Nazwa firmy">
            {offer.companyName}
          </DetailsItem>
          <DetailsItem md="3" xl="3" label="Lokalizacja">
            {offer.voivodeship}
          </DetailsItem>
          <DetailsItem md="3" xl="3" label="Ważne do">
            {new Date(offer.expirationDate).toLocaleDateString(undefined, {})}
          </DetailsItem>
        </Row>
        <IndexLinkContainer to={`/jobOffers/${offer.id}`}>
          <Button>Pokaż szczegóły</Button>
        </IndexLinkContainer>
      </Col>
    </Row>
  );
};

export default JobOfferInfo;
