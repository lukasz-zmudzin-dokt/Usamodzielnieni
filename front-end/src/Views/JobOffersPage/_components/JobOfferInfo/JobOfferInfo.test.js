import React from "react";
import { render } from "@testing-library/react";
import JobOfferInfo from "./JobOfferInfo";
import { MemoryRouter } from 'react-router-dom';

describe('JobOfferInfo', () => {
    it('should render without crashing', () => {
        const offer = {
            id: "123",
            title: "Tytu≈Ç oferty",
            companyName: "Nazwa firmy",
            voivodeship: "mazowieckie",
            expirationDate: "2020-12-24"
        };

        const { container } = render(
            <MemoryRouter>
                <JobOfferInfo offer={offer} />
            </MemoryRouter>
        );

        expect(container).toMatchSnapshot();
    });
});