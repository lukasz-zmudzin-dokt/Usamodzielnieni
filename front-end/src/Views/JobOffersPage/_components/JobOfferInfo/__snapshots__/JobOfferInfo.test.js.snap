// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobOfferInfo should render without crashing 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="col"
    >
      <h5>
        Tytuł oferty
      </h5>
      <div
        class="row"
      >
        <div
          class="detailsItem col-xl-3 col-md-4"
        >
          <div
            class="detailsItem__label"
          >
            Nazwa firmy
          </div>
          <div
            class="detailsItem__value"
          >
            Nazwa firmy
          </div>
        </div>
        <div
          class="detailsItem col-xl-3 col-md-4"
        >
          <div
            class="detailsItem__label"
          >
            Lokalizacja
          </div>
          <div
            class="detailsItem__value"
          >
            mazowieckie
          </div>
        </div>
        <div
          class="detailsItem col-xl-3 col-md-4"
        >
          <div
            class="detailsItem__label"
          >
            Ważne do
          </div>
          <div
            class="detailsItem__value"
          >
            2020-12-24
          </div>
        </div>
      </div>
      <a
        class="btn btn-primary"
        href="/jobOffers/123"
      >
        Pokaż szczegóły
      </a>
    </div>
  </div>
</div>
`;
